sdk/preview_termination_queries.go

```diff
@@ -0,0 +1,794 @@
+package sdk
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	corebig "math/big"
+
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/builtin/v9/miner"
+	"golang.org/x/exp/constraints"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-bitfield"
+	"github.com/filecoin-project/go-state-types/abi"
+	lotusapi "github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+	miner8 "github.com/filecoin-project/specs-actors/v8/actors/builtin/miner"
+	poolstypes "github.com/glifio/go-pools/types"
+	"github.com/glifio/go-pools/util"
+	"github.com/ipfs/go-cid"
+	cbor "github.com/ipfs/go-ipld-cbor"
+)
+
+// PreviewTerminateSector gets the burnt funds for when a single sector is terminated
+func (q *fevmQueries) PreviewTerminateSector(
+	ctx context.Context,
+	minerAddr address.Address,
+	tipset string,
+	vmHeight uint64,
```

Comment:
if we have the tipset, do we also then need the height?

Reply:
I think I'm always calling it with zero for the height, so the tipset parsing code will then lookup the height. I am usually supplying a tipset string like '@height'. This code is based on the Lotus CLI for `lotus state compute-state` which lets you supply a `--vm-height` parameter. I think this is so you can compute the state for different heights based on which fork the chain is on. We probably would never use it, but if we leave it in there, we can also do that.